In [10]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

# Use the kagglehub client library to attach Kaggle resources like competitions, datasets, and models to your session
# Learn more about kagglehub: https://github.com/Kaggle/kagglehub/blob/main/README.md

import kagglehub
# kagglehub.dataset_download('<owner>/<dataset-slug>')

/kaggle/input/datasets/aaronpole/superstore-clean/superstore_clean.csv


In [13]:

df = pd.read_csv('/kaggle/input/datasets/aaronpole/superstore-clean/superstore_clean.csv', encoding='latin-1')


Shape: (9994, 21)
Nulls: 0
   Row_ID        Order_ID  Order_Date   Ship_Date     Ship_Mode Customer_ID  \
0       1  CA-2013-152156  09-11-2013  12-11-2013  Second Class    CG-12520   
1       2  CA-2013-152156  09-11-2013  12-11-2013  Second Class    CG-12520   
2       3  CA-2013-138688  13-06-2013  17-06-2013  Second Class    DV-13045   

     Customer_Name    Segment        Country         City  ... Postal_Code  \
0      Claire Gute   Consumer  United States    Henderson  ...       42420   
1      Claire Gute   Consumer  United States    Henderson  ...       42420   
2  Darrin Van Huff  Corporate  United States  Los Angeles  ...       90036   

   Region       Product_ID         Category Sub_Category  \
0   South  FUR-BO-10001798        Furniture    Bookcases   
1   South  FUR-CH-10000454        Furniture       Chairs   
2    West  OFF-LA-10000240  Office Supplies       Labels   

                                        Product_Name   Sales  Quantity  \
0                  Bush Some

In [17]:
df['Order_Date'] = pd.to_datetime(df['Order_Date'],dayfirst=True)
df['Ship_Date']  = pd.to_datetime(df['Ship_Date'],dayfirst=True)

df['Days_To_Ship']  = (df['Ship_Date'] - df['Order_Date']).dt.days
df['Order_Year']    = df['Order_Date'].dt.year
df['Order_Month']   = df['Order_Date'].dt.month
df['Is_Loss']       = df['Profit'] < 0

df['Discount_Band'] = pd.cut(
    df['Discount'],
    bins=[-0.01, 0, 0.2, 1],
    labels=['No Discount', 'Low', 'High']
)

print(df[['Days_To_Ship', 'Order_Year', 'Is_Loss', 'Discount_Band']].head())
print("\nLoss-making orders:", df['Is_Loss'].sum())

   Days_To_Ship  Order_Year  Is_Loss Discount_Band
0             3        2013    False   No Discount
1             3        2013    False   No Discount
2             4        2013    False   No Discount
3             7        2012     True          High
4             7        2012    False           Low

Loss-making orders: 1871


## Key finding — Notebook 1
18.7% of all orders (1,871 out of 9,994) are loss-making.
Derived columns added: days_to_ship, order_year, order_month, is_loss, discount_band.
Data verified: 0 nulls, correct dtypes, ready for EDA.